### ISSUE TITLE
Installation: Windows SmartScreen

### ISSUE THREAD

**Issue by pedzed** [keeweb/keeweb#269]
Opening the 32-bit installer on Windows 10:

> Windows protected your PC
>
> Windows SmartScreen prevented an unrecognized app from starting. Running this app might put your PC at risk.

Perhaps because the publisher information is not set?

**Comment by antelle** [comment]
Yes, it's because currently we don't have code signing.
The same for mac: #196

**Comment by pedzed** [comment]
When do you plan to do it?

**Comment by antelle** [comment]
Once I have extra $100 per year for Apple Developer program, and then $170 per year for Windows certificate.

**Comment by pedzed** [comment]
OK, that's quite expensive. I hope that you can find a solution for it, whether it requires money or not.

_From: http://stackoverflow.com/a/16327654:_
> To quote from [MSDN's website][1]:
> 
> > Detractors may claim that SmartScreen is “forcing” developers to spend
> > money on certificates. It should be stressed that EV code signing
> > certificates are not required to build or maintain reputation with
> > SmartScreen. Files signed with standard code signing certificates and
> > even unsigned files continue to build reputation as they have since
> > Application Reputation was introduced in IE9 last year. However, the
> > presence of an EV code signing certificate is a strong indicator that
> > the file was signed by an entity that has passed a rigorous validation
> > process and was signed with hardware which allows our systems to
> > establish reputation for that entity more quickly than unsigned or
> > non-EV code signed programs.
> 
> In other words, EV (paid) validation is just one factor in a large algorithm that determines whether the SmartScreen warning is displayed or not. If you have a lot of people that download your program, or if your program download link has not changed in a while, with some work you can get your program **not** to show the warning. Also, by digitally signing your code, you can increase your Appication Reputation. This is straight from [Microsoft's webpage on the topic][2].
> 
> 
[1]: http://blogs.msdn.com/b/ie/archive/2012/08/14/microsoft-smartscreen-amp-extended-validation-ev-code-signing-certificates.aspx
[2]: http://ie.microsoft.com/testdrive/Browser/DownloadReputation/Default.html?o=1

Perhaps you could do something like fundraising. I do love KeeWeb and for myself I don't mind ignoring the warning (or building from source), but it does give a negative impression to people I want to recommend this great application.

**Comment by antelle** [comment]
I've investigated the issue (also please see linked issue 118 above, about Symantec blocking). Certificate is of course one step towards good reputation, but in our case, everything is worse than in usual apps because we're using Electron framework. So, it's just one exe for all other apps using it: electron can be used by malware as well. That's why anti-viruses and built-in filters can't just trust this executable and most probably will block it without a signature.
To sum up, code signing is mandatory here.
I understand that such warnings give us negative impressions. Certum provides some cheap alternatives, I'm thinking whether it's possible to use their cert.

**Comment by antelle** [comment]
I've asked about Certum certificates, and it looks like, they certificates are rather good. Sent a request for their cheap cert, let's see if it helps. Anyway, having this certificate is better than nothing.

**Comment by antelle** [comment]
Exe was signed. For now, only installer because it's technically difficult to sign the exe itself in this version. The next version with new installer will contain signed app as well. Now we have to wait a bit and gain some reputation, to make SmartScreen happy. Let's reopen the issue because now it's not just a duplicate of signing.

**Comment by antelle** [comment]
I've been testing this from time to time. Now SmartScreen doesn't show an alert for me. So, I assume it's fixed.

**Comment by pedzed** [comment]
> Now SmartScreen doesn't show an alert for me. So, I assume it's fixed.

Yes, I can confirm. I just downloaded it twice on Windows 10.
